core/arch/arm/kernel/boot.c

```diff
@@ -1184,7 +1192,11 @@ static void discover_nsec_memory(void)
 	const struct core_mmu_phys_mem *mem_begin = NULL;
 	const struct core_mmu_phys_mem *mem_end = NULL;
 	size_t nelems;
+#ifdef CFG_EXTERNAL_DT
 	void *fdt = get_external_dt();
+#else /* CFG_EXTERNAL_DT */
+	void *fdt = get_embedded_dt();
```

Comment:
this should be in a dedicated commit: get NSEC memory from embedded DT when CFG_EXTERNAL_DT is disabled.